DX-715: Resurrect Card Payments Direct (#258)
SwedbankPay/developer.swedbankpay.com: payments/card/direct.md

It is followed up by posting a capture, cancellation or reversal transaction.

An example of an abbreviated `POST` request is provided below. Each individual
Property of the JSON document is described in the following section.
An example of an expanded `POST` request is available in the
[other features section][purchase].

{% include alert-risk-indicator.md %}

{% include card-purchase.md %}

## Type of authorization - Intent

* **Authorization (two-phase):** If you want the credit card to reserve the
  amount, you will have to specify that the `intent` of the `Purchase` is
  `Authorization`. The amount will be reserved but not charged. You will later
  (i.e. when you are ready to ship the purchased products) have to make a
  [Capture][capture] or [Cancel][cancel] request.

### General

* *No 3-D Secure and card acceptance*: There are optional paramers that can be
  used in relation to 3-D Secure and card acceptance. By default, most credit
  card agreements with an acquirer will require that you use 3-D Secure for card
  holder authentication. However, if your agreement allows you to make a card
  payment without this authentication, or that specific cards can be declined,

>   you may adjust these optional parameters when posting in the payment.

Reviewer: ```suggestion
  you may adjust these optional parameters when creating the payment.
```
I found this sentence to be weirdly worded. `posting` is awkwardly technical term in a text that does not otherwise contain technical jargon.

  you will have to redirect the payer.
* When the payment is completed, the payer needs to be redirected back to your
  merchant/webshop site.
* Finally you make a `GET` request towards Swedbank Pay with the `paymentID`
  received in the first step, which will return the purchase result.

The sequence diagram below shows a high level description of a complete
purchase, and the requests you have to send to Swedbank Pay. The links will take
you directly to the corresponding API description.

```mermaid
sequenceDiagram
    participant Payer
    participant Merchant
    participant SwedbankPay as Swedbank Pay

    activate Payer
    Payer->>+Merchant: start purchase
    deactivate Payer
    Merchant->>+SwedbankPay: POST /psp/creditcard/payments
    deactivate Merchant
    note left of Merchant: First API Request
    SwedbankPay-->>+Merchant: rel: direct-authorization
    deactivate SwedbankPay
    Merchant->>+SwedbankPay: GET <payment.id>
    deactivate Merchant
    note left of Merchant: Second API request
    SwedbankPay-->>+Merchant: rel: direct-authorization
    deactivate SwedbankPay
    Merchant-->>-Payer: display purchase result
```

## API Requests

> The API requests are displayed in the [purchase flow](#purchase-flow-mobile).

Reviewer: ```suggestion
The API requests are displayed in the [purchase flow][#purchase-flow-mobile].
```
Tested link in GitHub view and it does not work, it also does not follow the convention of the rest of the documentation of this system.

You can [create a card `payment`][create-payment] with following `operation`
options:

> * [Purchase][purchase]

Reviewer: ```suggestion
* [`Purchase`][purchase]
```
Isn't this and the other entries on this list refering to a code element, and should thus be marked as code elements?

* [Recur][recur]
* [Payout][payout]
* [Verify][verify]

Our `payment` example above uses the [`purchase`][purchase] value which is
explained below.

### Purchase

A `Purchase` payment is a straightforward way to charge the card of the payer.
It is followed up by posting a capture, cancellation or reversal transaction.

An example of an abbreviated `POST` request is provided below. Each individual
Property of the JSON document is described in the following section.
An example of an expanded `POST` request is available in the
[other features section][purchase].

{% include alert-risk-indicator.md %}

{% include card-purchase.md %}

## Type of authorization - Intent

* **Authorization (two-phase):** If you want the credit card to reserve the

>   amount, you will have to specify that the `intent` of the `purchase` is

Reviewer: ```suggestion
  amount, you will have to specify that the `intent` of the `Purchase` is
```
`purchase` is uppercased in the rest of the document.

  `Authorization`. The amount will be reserved but not charged. You will later
  (i.e. when you are ready to ship the purchased products) have to make a
  [Capture][Capture] or [Cancel][Cancel] request.

### General

* *No 3-D Secure and card acceptance*: There are optional paramers that can be
  used in relation to 3-D Secure and card acceptance. By default, most credit
  card agreements with an acquirer will require that you use 3-D Secure for card
  holder authentication. However, if your agreement allows you to make a card
  payment without this authentication, or that specific cards can be declined,
  you may adjust these optional parameters when posting in the payment.
* *Defining CallbackURL*: When implementing a scenario, it is optional to set
  a [CallbackURL][callback] in the `POST` request. If `CallbackURL` is set, then
  Swedbank Pay will send a postback request to this URL when the consumer has
  fulfilled the payment. [See the Callback API description here][callback].

## Type of capture - Intent

* **AutoCapture (one-phase):** If you want the credit card to be charged right

>   away, you will have to specify that the `intent` of the `purchase` is

Reviewer: ```suggestion
  away, you will have to specify that the `intent` of the `Purchase` is
```
`purchase` is uppercased in the rest of the document.

  Payer->>+Merchant: start purchase
  deactivate Payer
  Merchant->>+SwedbankPay: POST /psp/creditcard/payments
  deactivate Merchant
  note left of Merchant: First API request
  SwedbankPay-->+Merchant: payment resource
  deactivate SwedbankPay
  Merchant-->>+SwedbankPay: GET <payment.id>
  deactivate Merchant
  note left of Merchant: Second API request
  SwedbankPay-->>+Merchant: rel: direct-authorization
  deactivate SwedbankPay
  Merchant-->>+Payer: redirect to 3-D Secure page
  deactivate Merchant
  Payer->>+IssuingBank: 3-D Secure authentication
  deactivate Payer
  note left of Payer: redirect to card issuing bank
  IssuingBank-->>+Payer: redirect to merchant
  deactivate IssuingBank
  note left of Payer: redirect back to merchant
  Payer->>+Merchant: access merchant page
  deactivate Payer
  Merchant->>+SwedbankPay: GET <payment.id>
  deactivate Merchant
  note left of Merchant: Third API request
  SwedbankPay-->>+Merchant: rel: redirect-authorization
  deactivate SwedbankPay
  Merchant-->>Payer: display purchase result
  deactivate Merchant
```

## Options after posting a purchase payment

* If the payment shown above is done as a two-phase (Authorization), you will

>   need to implement the [Capture][Capture] and [Cancel][Cancel] requests.

Reviewer: ```suggestion
  need to implement the [Capture][capture] and [Cancel][cancel] requests.
```
Links are not uppercase except for abbriviations.

* **Abort:** It is possible to [abort a payment][abort] if the payment has no
  successful transactions.
* For `reversals`, you will need to implement the [Reversal][reversal] request.
* **Callback from Swedbank Pay:** Whenever changes to the payment occur a
  [Callback request][callback] will be posted to the `callbackUrl`, generated
  when the payment was created.

{% include iterator.html prev_href="seamless-view" prev_title="Back: Seamless View"
next_href="after-payment" next_title="Next: After Payment" %}

[abort]: /payments/card/other-features/#abort
[expansion]: /payments/card/other-features/#expansion
[callback]: /payments/card/other-features/#callback

> [Cancel]: /payments/card/after-payment/#Cancellations

Reviewer: ```suggestion
[cancel]: /payments/card/after-payment/#Cancellations
```

> [Capture]: /payments/cardd/after-payment/#Capture

Reviewer: ```suggestion
[capture]: /payments/cardd/after-payment/#Capture
```
